src/runner.rs

```diff
@@ -102,8 +102,21 @@ pub fn run(av: ArgValues) -> error::Result<()> {
     });
 
     let event_loop = EventLoop::new();
+
+    let shader_name = av.getid.as_ref().or_else(|| av.examplename.as_ref());
```

Comment:
This doesn't work properly with shaders that are created/run locally, rather than downloaded automatically from shadertoy. You'll need to first check if `av.shaderpath` is `Some` and use that name if so, falling back to this existing logic otherwise.

Reply:
How should I determine the name from `av.shaderpath`? Should I just use `av.shaderpath` verbatim?

Reply:
Yeah, `av.shaderpath` directly should work

Reply:
updated!